thrust/detail/config/cpp_dialect.h

```diff
@@ -14,19 +14,37 @@
  *  limitations under the License.
  */
 
+/*! \file cpp_dialect.h
+ *  \brief Detect the version of the C++ standard used by the compiler.
+ */
+
 #pragma once
 
-#if   __cplusplus < 201103L
-  #define THRUST_CPP03
-  #define THRUST_CPP_DIALECT 2003
-#elif __cplusplus < 201402L
-  #define THRUST_CPP11
-  #define THRUST_CPP_DIALECT 2011
-#elif __cplusplus < 201703L
-  #define THRUST_CPP14
-  #define THRUST_CPP_DIALECT 2014
-#else
-  #define THRUST_CPP17
-  #define THRUST_CPP_DIALECT 2017
-#endif
+#include <thrust/detail/config/deprecated.h>
+
+#ifndef THRUST_CPP_DIALECT
+
+// MSVC does not define __cplusplus correctly. _MSVC_LANG is used instead
+// (MSVC 2015U3+ only)
+#  ifdef _MSVC_LANG
+#    define THRUST___CPLUSPLUS _MSVC_LANG
```

Comment:
Rename, this name is reserved for implementations of ISO C++ (two consecutive underscores).

Reply:
Will do. We have other macros that violate this rule, but I agree we shouldn't be adding more.